rust/routee-compass-py/src/app_graph_ops.rs

```diff
@@ -0,0 +1,98 @@
+use pyo3::{exceptions::PyException, PyErr, PyResult};
+use routee_compass::app::search::search_app_graph_ops::SearchAppGraphOps;
+use routee_compass_core::{
+    algorithm::search::direction::Direction,
+    model::road_network::{edge_id::EdgeId, vertex_id::VertexId},
+    util::unit::{as_f64::AsF64, DistanceUnit},
+};
+
+use crate::app_wrapper::CompassAppWrapper;
+
+pub fn graph_edge_origin(app: &CompassAppWrapper, edge_id: usize) -> PyResult<usize> {
+    let edge_id_internal = EdgeId(edge_id);
+    app.routee_compass
+        .search_app
+        .get_edge_origin(edge_id_internal)
+        .map(|o| o.0)
+        .map_err(|e| {
+            PyException::new_err(format!(
+                "error retrieving edge origin for edge_id {}: {}",
+                edge_id, e
+            ))
+        })
+}
+
+pub fn graph_edge_destination(app: &CompassAppWrapper, edge_id: usize) -> PyResult<usize> {
+    let edge_id_internal = EdgeId(edge_id);
+    app.routee_compass
+        .search_app
+        .get_edge_destination(edge_id_internal)
+        .map(|o| o.0)
+        .map_err(|e| {
+            PyException::new_err(format!(
+                "error retrieving edge destination for edge_id {}: {}",
+                edge_id, e
+            ))
+        })
+}
+
+pub fn graph_edge_distance(
+    app: &CompassAppWrapper,
+    edge_id: usize,
+    distance_unit: Option<String>,
+) -> PyResult<f64> {
+    let du_internal_result: PyResult<Option<DistanceUnit>> = match distance_unit {
+        Some(du_str) => {
+            let mut enquoted = du_str.clone();
```

Comment:
Wonder if we could add some comments here to explain why this has to be done or maybe create a simple ops function that does this?

Reply:
for real. i was fishing around for this trick and have run into this previously as well. i'll add a comment, as i'm not sure it's the best way to leverage serde to decode non-parameterized enums from a string.

Reply:
eh, @nreinicke, i changed my mind; i'll make a generic utility for it, and then leverage that in FromStr implementations for the Unit types. coming right up.